src/thermo/ThermoPhase.cpp

```diff
@@ -194,6 +194,7 @@ void ThermoPhase::setState_HP(double Htarget, double p, double rtol)
 
 void ThermoPhase::setState_UV(double u, double v, double rtol)
 {
+    assertCompressible("setState_UV");
```

Comment:
Why not just let this through? An exception will be raised once it tries to set the density inside `setState_HPorUV`, and that avoids running this check twice.

Reply:
I added it here as it triggers a more meaningful error (and there's no point in going farther as `setDensity` will fail no matter what).